src/views/pages/Inbox/index.js

```diff
@@ -1,17 +1,19 @@
 import React, {useEffect, useState} from 'react';
-import MessageService from '../../../services/Message.service';
+import MessageAPIService from '../../../services/MessageAPI';
 import ListView from '../../components/ListView';
 
-const message = new MessageService();
+const message = new MessageAPIService();
 
 function Inbox() {
   const [messageList, setMessageList] = useState([]);
 
   useEffect(() => {
-    const sub = message.messages$.subscribe((d) => {
+    const sub = message.received_messages$.subscribe((d) => {
       setMessageList(d);
     });
 
+    message.checkMessages(1);
```

Comment:
@isaacadams do you know off the top of your head what the best way to populate this on load and then periodically/manually refresh would be?

Reply:
useEffect() can be used like onMount used to be used, I believe.

Reply:
But that would only cover the initial load, I guess.

Reply:
Yeah, I had that thought too. I'd really like to do a situation where we can 'pull down to refresh' when we get to mobile but my Googling so far has been in vain. :joy:

Reply:
populate on load
```jsx
useEffect(() => {
   ....
}, []);
```

the important thing is the empty array being passed in. That means it will only run once when the component mounts and the cleanup will run when it un-mounts.

in order to run the logic within the useEffect again, you will want to add the variables into the array that the effect should "react" to changing

so when the count variable changes, the below effect will run again 
```jsx
const [count, setCount] = useState(0);
useEffect(() => {
   console.log(count);
}, [count]);
```

Reply:
if you want the effect to run literally periodically, like based on a timer, then you can use the following

```jsx
useEffect(() => {
  let id = setInterval(callback, delay);
  return () => clearInterval(id);
}, []);
```